### PR TITLE
feat(AwsProvider): include new structure for AWS provider

### PR DESCRIPTION
**Pull request by n4ch04**
### Context

Please include relevant motivation and context for this PR.

### Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/__main__.py [hunk 1]
```diff
@@ -39,7 +39,8 @@
 from prowler.lib.outputs.outputs import extract_findings_statistics
 from prowler.lib.outputs.slack import send_slack_message
 from prowler.lib.outputs.summary_table import display_summary_table
-from prowler.providers.aws.aws_provider import get_available_aws_service_regions
+
+# from prowler.providers.aws.aws_provider_new import get_available_aws_service_regions
```
Comment (jfagoagas) [inline prowler/__main__.py]:
```suggestion
```

File: prowler/providers/aws/aws_provider_new.py [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+    validate_aws_credentials,
+)
+from prowler.providers.aws.lib.organizations.organizations import (
+    get_organizations_metadata,
+)
+from prowler.providers.common.provider import CloudProvider
+
+
+@dataclass
+class AWS_Organizations_Info:
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider_new.py]: Can you please change that to `AWSOrganizationsInfo`? To keep Python naming standards.

File: prowler/providers/aws/aws_provider_new.py [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+    account_details_email: str
+    account_details_name: str
+    account_details_arn: str
+    account_details_org: str
+    account_details_tags: str
+
+
+@dataclass
+class AWS_Credentials:
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider_new.py]: Can you please change that to `AWSCrdentials`? To keep Python naming standards.

File: prowler/providers/aws/aws_provider_new.py [hunk 4]
```diff
@@ -0,0 +1,622 @@
...
+    aws_access_key_id: str
+    aws_session_token: str
+    aws_secret_access_key: str
+    expiration: datetime
+
+
+@dataclass
+class AWS_Assume_Role:
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider_new.py]: Can you please change that to `AWSAssumeRole`? To keep Python naming standards.

File: prowler/providers/aws/aws_provider_new.py [hunk 5]
```diff
@@ -0,0 +1,622 @@
...
+    )
+    organizations_metadata: AWS_Organizations_Info = AWS_Organizations_Info(
+        account_details_email=None,
+        account_details_name=None,
+        account_details_arn=None,
+        account_details_org=None,
+        account_details_tags=None,
+    )
+    audit_resources: Optional[Any] = None
+    audit_metadata: Optional[Any] = None
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider_new.py]: Why `Optional[Any] = None`? `Optional` is the same as `Union[Type|None]`

File: prowler/providers/aws/aws_provider_new.py [hunk 6]
```diff
@@ -0,0 +1,622 @@
...
+    audit_config: dict = {}
+    mfa_enabled: bool = False
+    ignore_unused_services: bool = False
+
+    def __init__(self, args):
+        logger.info("Setting AWS provider ...")
+        # Parse input arguments
+        # Assume Role Options
+        arguments = args.__dict__
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider_new.py]: I think you don't need that if you are receiving the `args` as an argparse `Namespace`.

Also all the `get()` below can be changed, e.g.: `arguments.role`.

File: prowler/providers/aws/aws_provider_new.py [hunk 7]
```diff
@@ -0,0 +1,622 @@
...
+            self.organizations_metadata = get_organizations_metadata(
+                self.identity.account, self.assumed_role.assumed_role_credentials
+            )
+            logger.info("Organizations metadata retrieved")
+        if self.session.session.region_name:
+            self.identity.profile_region = self.session.session.region_name
+        else:
+            self.identity.profile_region = "us-east-1"
+
+        if not arguments.get("only_logs"):
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider_new.py]: I think this should be out of the init. One of the main ideas to make a generic provider is to only call the provider `print_credentials` in one place without the need to include that in the `init`.

File: prowler/providers/aws/aws_provider_new.py [hunk 8]
```diff
@@ -0,0 +1,622 @@
...
+            self.print_credentials()
+
+        # Parse Scan Tags
+        if arguments.get("resource_tags"):
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider_new.py]: The same as above.

File: prowler/providers/aws/aws_provider_new.py [hunk 9]
```diff
@@ -0,0 +1,622 @@
...
+            input_resource_tags = arguments.get("resource_tags")
+            self.audit_resources = self.get_tagged_resources(input_resource_tags)
+
+        # Parse Input Resource ARNs
+        if arguments.get("resource_arn"):
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider_new.py]: The same as above.

File: prowler/providers/aws/aws_provider_new.py (hunk 7 above)
Comment (jfagoagas) [reply prowler/providers/aws/aws_provider_new.py]: Right now the class CloudProvider does nothing
```
class CloudProvider(ABC):
    @abstractmethod
    def setup_session(self):
        pass

    @abstractmethod
    def print_credentials(self):
        pass

    def validate_arguments(self):
        pass
```

This needs to be the one in charge to execute the logic to instantiate the provider.

Comment (jfagoagas) [reply prowler/providers/aws/aws_provider_new.py]: And we should rename `CloudProvider` to `Provider`,

**Review by jfagoagas — CHANGES_REQUESTED** [review]
We are close to have the new providers 🎉 

Please check the comments and let me know if you need to clarify something, great job 👏

File: prowler/providers/aws/aws_provider_new.py (hunk 3 above)
Comment (MrCloudSec) [reply prowler/providers/aws/aws_provider_new.py]: Done

File: prowler/providers/aws/aws_provider_new.py (hunk 4 above)
Comment (MrCloudSec) [reply prowler/providers/aws/aws_provider_new.py]: Done

File: prowler/providers/aws/aws_provider_new.py (hunk 2 above)
Comment (MrCloudSec) [reply prowler/providers/aws/aws_provider_new.py]: Done

File: prowler/__main__.py (hunk 1 above)
Comment (MrCloudSec) [reply prowler/__main__.py]: Done

File: prowler/providers/aws/aws_provider_new.py (hunk 6 above)
Comment (MrCloudSec) [reply prowler/providers/aws/aws_provider_new.py]: done

File: prowler/providers/aws/aws_provider_new.py (hunk 8 above)
Comment (MrCloudSec) [reply prowler/providers/aws/aws_provider_new.py]: done

File: prowler/providers/aws/aws_provider_new.py (hunk 9 above)
Comment (MrCloudSec) [reply prowler/providers/aws/aws_provider_new.py]: done

File: prowler/providers/aws/aws_provider_new.py (hunk 5 above)
Comment (MrCloudSec) [reply prowler/providers/aws/aws_provider_new.py]: Done

File: prowler/providers/aws/aws_provider_new.py (hunk 7 above)
Comment (MrCloudSec) [reply prowler/providers/aws/aws_provider_new.py]: name changed, can we do the generic calls in other PR?

**Review by jfagoagas — APPROVED** [review]
🤩  One step closer from the finish line 🏁 

Thanks @sergargar and @n4ch04 🚀

**Pull request merged**